src/include/snapper/dialogs.rb

```diff
@@ -1233,6 +1018,196 @@ def ShowDialog
+    def ReadDialog
+      return :abort if !Confirm.MustBeRoot
+
+      Wizard.RestoreHelp(Ops.get_string(@HELPS, "read", ""))
+      ret = Snapper.Init()
+      ret ? :next : :abort
+    end
+
+    def open_modify_dialog(content)
+      UI.OpenDialog(
+        Opt(:decorated),
+        HBox(
+          HSpacing(1),
+          VBox(
+            VSpacing(0.5),
+            HSpacing(65),
+            content,
+            VSpacing(0.5),
+            ButtonBox(
+              PushButton(Id(:ok), Label.OKButton),
+              PushButton(Id(:cancel), Label.CancelButton)
+            ),
+            VSpacing(0.5)
+          ),
+          HSpacing(1)
+        )
+      )
+    end
+
+    def snapshot_term(prefix, data)
+      data = deep_copy(data)
+      HBox(
+        HSpacing(),
+        Frame(
+          "",
+          HBox(
+            HSpacing(0.4),
+            VBox(
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "description")),
+                Opt(:hstretch),
+                _("Description"),
+                Ops.get_string(data, "description", "")
+              ),
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "userdata")),
+                Opt(:hstretch),
+                _("User data"),
+                Snapper.userdata_to_string(data["userdata"])
+              ),
+              Left(
+                ComboBox(
+                  Id(Ops.add(prefix, "cleanup")),
+                  Opt(:editable, :hstretch),
+                  # combo box label
+                  _("Cleanup algorithm"),
+                  cleanup_items(Ops.get_string(data, "cleanup", ""))
+                )
+              )
+            ),
+            HSpacing(0.4)
+          )
+        ),
+        HSpacing()
+      )
+    end
+
+    def update_snapshots
+      # busy popup message
+      Popup.ShowFeedback("", _("Reading list of snapshots..."))
+
+      Snapper.ReadSnapshots()
+
+      Popup.ClearFeedback
+
+      UI.ChangeWidget(Id(:snapshots_table), :Items, get_snapshot_items)
+      selected = UI.QueryWidget(Id(:snapshots_table), :SelectedItems) || []
+      enable_buttons([:modify, :show, :delete], selected.size == 1)
+      enable_buttons([:delete], selected.size >= 1)
+
+    end
+
+    def get_snapshot_items
+      snapshot_items = []
+
+      Snapper.snapshots.each_with_index do |s,i|
+        num = s["num"] || 0
+        start_date = (num != 0) ? timestring(s["date"]) : ""
+        end_date = ""
+        userdata = Snapper.userdata_to_string(s["userdata"])
+        desc = s["description"].to_s
+        if s["type"] == :SINGLE
+          type = _("Single")
+        elsif s["type"] == :POST
+          pre = s["pre_num"] || 0 # pre canot be 0
+          index = Ops.get(Snapper.id2index, pre, -1)
+          if pre == 0 || index == -1
+            Builtins.y2warning(
+              "something wrong - pre:%1, index:%2",
```

Comment:
I think better to write to log whole snapshot so something like "something wrong - #{s.inspect}"

BTW for new code is suggested to use object logger - https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/y2logger.rb#L47